src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -175,6 +175,13 @@ CandidateReason.Ambiguous or
                 }
             }
 
+            if (name is IdentifierNameSyntax { Identifier.Text: "args" } &&
+                symbol is IParameterSymbol { ContainingSymbol: IMethodSymbol { Name: WellKnownMemberNames.TopLevelStatementsEntryPointMethodName } })
```

Comment:
Is the first check necessary? I think it can be moved as an assert inside the condition. It should be okay either way.

Reply:
i thought about that, but i like the code being explicit about the precise case it is handling.